src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IConversionExpression.cs

```diff
@@ -3129,6 +3129,1829 @@ public int M1()
+
+            VerifyOperationTreeAndDiagnosticsForTest<VariableDeclaratorSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void ConversionExpression_Explicit_UnboxingValueTypeFromInterfaceConversion_InvalidNoConversion()
+        {
+            string source = @"
+using System;
+
+class C1
+{
+    void M1()
+    {
+        I1 i = null;
+        S1 /*<bind>*/s1 = (S1)i/*</bind>*/;
+    }
+}
+
+interface I1 { }
+
+struct S1 { }
+";
+            string expectedOperationTree = @"
+IVariableDeclarationStatement (1 declarations) (OperationKind.VariableDeclarationStatement, IsInvalid) (Syntax: 'S1 /*<bind> ... *</bind>*/;')
+  IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration, IsInvalid) (Syntax: 'S1 /*<bind> ... *</bind>*/;')
+    Variables: Local_1: S1 s1
+    Initializer: IConversionExpression (ConversionKind.Invalid, Explicit) (OperationKind.ConversionExpression, Type: S1, IsInvalid) (Syntax: '(S1)i')
+        ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: I1) (Syntax: 'i')
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS0030: Cannot convert type 'I1' to 'S1'
+                //         S1 /*<bind>*/s1 = (S1)i/*</bind>*/;
+                Diagnostic(ErrorCode.ERR_NoExplicitConv, "(S1)i").WithArguments("I1", "S1").WithLocation(9, 27)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<VariableDeclaratorSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void ConversionExpression_Explicit_UnboxingVarianceConversion()
+        {
+            string source = @"
+using System;
+using System.Collections.Generic;
+
+class C1
+{
+    void M1()
+    {
+        IList<I1> i1List = new List<I1>();
+        IList<S1> /*<bind>*/s1List = (IList<S1>)i1List/*</bind>*/;
+    }
+}
+
+interface I1 { }
+
+struct S1 : I1 { }
+";
+            string expectedOperationTree = @"
+IVariableDeclarationStatement (1 declarations) (OperationKind.VariableDeclarationStatement) (Syntax: 'IList<S1> / ... *</bind>*/;')
+  IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration) (Syntax: 'IList<S1> / ... *</bind>*/;')
+    Variables: Local_1: System.Collections.Generic.IList<S1> s1List
+    Initializer: IConversionExpression (ConversionKind.Cast, Explicit) (OperationKind.ConversionExpression, Type: System.Collections.Generic.IList<S1>) (Syntax: '(IList<S1>)i1List')
+        ILocalReferenceExpression: i1List (OperationKind.LocalReferenceExpression, Type: System.Collections.Generic.IList<I1>) (Syntax: 'i1List')
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<VariableDeclaratorSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void ConversionExpression_Explicit_TypeParameterSubclassConversion()
```

Comment:
`U` can't be a subclass of `T`, because `T` is a subclass of `U`. The test is testing what conversion is created from assigning a variable of type `U` to a variable of type `T`, as `U` is a subclass of `T`.